File: spec/asciidoc_spec.rb
# frozen_string_literal: true

require 'spec_helper'

RSpec.describe 'AsciiDoc' do
  it 'maintains indentation in ascii doc bullets' do
    test_bullet_inserted('rats', <<-INIT, <<-EXPECTED)
      = Pets!
      * dogs
      ** cats
    INIT
      = Pets!
      * dogs
      ** cats
      ** rats
    EXPECTED
  end

  it 'supports dot bullets' do
    test_bullet_inserted('cats', <<-INIT, <<-EXPECTED)
      = Pets!
      . dogs
    INIT
      = Pets!
      . dogs
      . cats
    EXPECTED
  end

  it 'supports nested dot bullets' do
    pending('FIXME: this test fails, but the functionality works')
    test_bullet_inserted('rats', <<-INIT, <<-EXPECTED)
      = Pets!
      . dogs
      .. cats
    INIT
      = Pets!
      . dogs
      .. cats
      .. rats
    EXPECTED
  end
end

File: test/asciidoc_spec.lua
local helpers = require("test.helpers")

describe("AsciiDoc", function()
  it("maintains indentation in ascii doc bullets", function()
    helpers.test_bullet_inserted(
      "rats",
      { "= Pets!", "* dogs", "** cats" },
      { "= Pets!", "* dogs", "** cats", "** rats" }
    )
  end)

  it("supports dot bullets", function()
    helpers.test_bullet_inserted("cats", { "= Pets!", ". dogs" }, { "= Pets!", ". dogs", ". cats" })
  end)

  it("supports nested dot bullets", function()
    helpers.test_bullet_inserted(
      "rats",
      { "= Pets!", ". dogs", ".. cats" },
      { "= Pets!", ". dogs", ".. cats", ".. rats" }
    )
  end)
end)
